core/src/utils/overlays.ts

```diff
@@ -54,7 +54,7 @@ export const createOverlay = <T extends HTMLIonOverlayElement>(tagName: string,
        * Convert the passed in overlay options into props
        * that get passed down into the new overlay.
        */
-      Object.assign(element, { ...opts });
+      Object.assign(element, { ...opts, hasController: true });
```

Comment:
Does the `hasController` prop need to be added to other overlay components besides `modal` and `popover`? Or is the issue specific to those two? This function gets used in all overlays from what I can tell.

Reply:
Not yet - the other overlays can't be used declaratively at the moment. Once they are we should add the prop though.